typespec-tests/src/main/java/com/client/naming/ClientModelClient.java

```diff
@@ -15,25 +15,25 @@
 import com.azure.core.http.rest.RequestOptions;
 import com.azure.core.http.rest.Response;
 import com.azure.core.util.BinaryData;
-import com.client.naming.implementation.ModelsImpl;
+import com.client.naming.implementation.ClientModelsImpl;
 import com.client.naming.models.ClientModel;
 import com.client.naming.models.JavaModel;
 
 /**
  * Initializes a new instance of the synchronous NamingClient type.
  */
 @ServiceClient(builder = NamingClientBuilder.class)
-public final class ModelClient {
+public final class ClientModelClient {
     @Generated
```

Comment:
curious why the client name changed？

Reply:
https://github.com/Azure/autorest.java/pull/2821#discussion_r1670368435

looks like tcgc bug fix